client/landing/stepper/declarative-flow/assembler-first-flow.ts

```diff
@@ -0,0 +1,264 @@
+
+			switch ( _currentStep ) {
+				case 'check-sites': {
+					// Check for unlaunched sites
+					if ( providedDependencies?.filteredSitesCount === 0 ) {
+						// No unlaunched sites, redirect to new site creation step
+						return navigate( 'site-creation-step' );
+					}
+					// With unlaunched sites, continue to new-or-existing-site step
+					return navigate( 'new-or-existing-site' );
+				}
+
+				case 'new-or-existing-site': {
+					if ( 'new-site' === providedDependencies?.newExistingSiteChoice ) {
+						return navigate( 'site-creation-step' );
+					}
+					return navigate( 'site-picker' );
+				}
+
+				case 'site-creation-step': {
+					return navigate( 'processing' );
+				}
+
+				case 'site-picker': {
+					return handleSelectSite( providedDependencies );
+				}
+
+				case 'processing': {
+					if ( results.some( ( result ) => result === ProcessingResult.FAILURE ) ) {
+						return navigate( 'error' );
+					}
+
+					// If we just created a new site, navigate to the assembler step.
+					if ( providedDependencies?.siteSlug && ! providedDependencies?.blogLaunched ) {
+						return handleSelectSite( {
+							...providedDependencies,
+							isNewSite: 'true',
+						} );
+					}
+
+					const params = new URLSearchParams( {
+						canvas: 'edit',
+						assembler: '1',
+					} );
+
+					return exitFlow( `/site-editor/${ siteSlug }?${ params }` );
+				}
+
+				case 'patternAssembler': {
+					return navigate( 'processing' );
+				}
+			}
+		};
+
+		const goBack = () => {
+			switch ( _currentStep ) {
+				case 'site-picker': {
+					return navigate( 'new-or-existing-site' );
+				}
+
+				case 'patternAssembler': {
+					return window.location.assign( `/themes/${ siteSlug }` );
+				}
+			}
+		};
+
+		const goNext = async () => {
+			switch ( _currentStep ) {
+				case 'launchpad':
+					skipLaunchpad( {
+						checklistSlug: ASSEMBLER_FIRST_FLOW,
+						siteId,
+						siteSlug,
+					} );
+					return;
+			}
+		};
+
+		return { submit, goBack, goNext };
+	},
+
+	useAssertConditions(): AssertConditionResult {
+		const flowName = this.name;
+		const isLoggedIn = useSelector( isUserLoggedIn );
+		const currentUserSiteCount = useSelector( getCurrentUserSiteCount );
+		const currentPath = window.location.pathname;
+		const isSiteCreationStep =
+			currentPath.endsWith( `setup/${ flowName }` ) ||
+			currentPath.endsWith( `setup/${ flowName }/` ) ||
+			currentPath.includes( `setup/${ flowName }/check-sites` );
+		const userAlreadyHasSites = currentUserSiteCount && currentUserSiteCount > 0;
+
+		// There is a race condition where useLocale is reporting english,
+		// despite there being a locale in the URL so we need to look it up manually.
+		// We also need to support both query param and path suffix localized urls
+		// depending on where the user is coming from.
+		const useLocaleSlug = useLocale();
+		// Query param support can be removed after dotcom-forge/issues/2960 and 2961 are closed.
```

Comment:
I see that both issues are closed now 😄

Reply:
Let's leave it as it is now and clean it up in the follow-up PR 😆